### PR TITLE
9 external file location

### PR DESCRIPTION
**Pull request by florian-dacosta**
Depend on https://github.com/OCA/server-tools/pull/515

This module allow to export file from ir.attachment.metadata or import files creating a ir.attachment.metadata.

Other modules could be added to implement action depending on the imported file.

Tests should be green after PR 515 is merged.

### CODE REVIEW

File: external_file_location/README.rst [hunk 1]
```diff
@@ -0,0 +1,60 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+    :alt: License
+
+External File Location
```
Comment (angelmoya) [inline external_file_location/README.rst]: Use last version of template
https://raw.githubusercontent.com/OCA/maintainer-tools/master/template/module/README.rst

File: external_file_location/README.rst [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+Installation
+============
+
+To install this module, you need to:
+
+* fs python module
+* Paramiko python module
+
+Usage
```
Comment (angelmoya) [inline external_file_location/README.rst]: Add runbot button

File: external_file_location/models/location.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+from openerp import models, fields, api
+from openerp.addons.external_file_location.tasks.filestore import FileStoreTask
+from openerp.addons.external_file_location.tasks.ftp import FtpTask
+from openerp.addons.external_file_location.tasks.sftp import SftpTask
+
+
+class Location(models.Model):
+    _name = 'external.file.location'
+    _description = 'Description'
```
Comment (angelmoya) [inline external_file_location/models/location.py]: Description -> Location

File: external_file_location/models/task.py [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+                                'to import all txt files')
+    filepath = fields.Char(help='Path to imported/exported file')
+
+    location_id = fields.Many2one('external.file.location', string='Location',
+                                  required=True)
+
+    attachment_ids = fields.One2many('ir.attachment.metadata', 'task_id',
+                                     string='Attachment')
+
+    move_path = fields.Char(string='Move path',
```
Comment (angelmoya) [inline external_file_location/models/task.py]: Move path -> Move Path

File: external_file_location/models/task.py [hunk 5]
```diff
@@ -0,0 +1,212 @@
...
+                            help='Imported File will be moved to this path')
+
+    new_name = fields.Char(string='New name',
```
Comment (angelmoya) [inline external_file_location/models/task.py]: New name -> New Name

File: external_file_location/models/task.py [hunk 6]
```diff
@@ -0,0 +1,212 @@
...
+                                    help='Action after import a file')
+
+    company_id = fields.Many2one(
+        'res.company', 'Company',
+        default=lambda self: self.env['res.company']._company_default_get(
+            'external.file.task'))
+
+    file_type = fields.Selection(
+        selection=[],
+        string="File type",
```
Comment (angelmoya) [inline external_file_location/models/task.py]: File type -> File Type

**Review by angelmoya — COMMENTED** [review]
I will test after pending PR will be aproved

**Comment by lasley** [comment]
@florian-dacosta - Can you please rebase on 9 in order to remove the already merged commits?

**Comment by bguillot** [comment]
I have tested it, it works well 
:+1:

**Comment by mourad-ehm** [comment]
:+1:

**Review by moylop260 — APPROVED** [review]

**Pull request merged**